src/main/webapp/js/issues-detail.js

```diff
@@ -48,116 +38,49 @@
-        if (orderBy && orderDirection) {
-            var order = [orderBy, orderDirection];
-            try {
-                dataTable.order(order).draw();
-            }
-            catch (ignore) { // TODO: find a way to determine the number of columns here
-                dataTable.order([[1, 'asc']]).draw();
-            }
-        }
-        // Store paging size
-        $(this).on('length.dt', function (e, settings, len) {
-            localStorage.setItem(id + '#table-length', len);
-        });
-        var storedLength = localStorage.getItem(id + '#table-length');
-        if ($.isNumeric(storedLength)) {
-            dataTable.page.len(storedLength).draw();
-        }
-    });
-
     /**
      * Activate tooltips.
      */
     $(function () {
         $('[data-toggle="tooltip"]').tooltip();
     });
 
-    /**
-     * Initializes the pie chart. There are two different variants: on small screens we have a carousel that
-     * switches between the two charts. On large screens both pie charts are drawn side-by-side.
-     */
-    function initializePieCharts () {
-        /**
-         * Creates a doughnut chart that shows the number of issues per severity.
-         * Requires that DOM <div> elements exist with the IDs '#severities-chart', '#single-severities-chart'.
-         */
-        view.getSeverityModel(function (severityModel) {
-            $('#severities-chart').renderPieChart(severityModel.responseJSON, true); // small screens
-            $('#single-severities-chart').renderPieChart(severityModel.responseJSON, false); // large screens
-        });
-        /**
-         * Creates a doughnut chart that shows the number of new, fixed and outstanding issues.
-         * Requires that DOM <div> elements exist with the IDs '#trend-chart', '#single-trend-chart'.
-         */
-        view.getTrendModel(function (pieModel) {
-            $('#trend-chart').renderPieChart(pieModel.responseJSON, true);
-            $('#single-trend-chart').renderPieChart(pieModel.responseJSON, false);
-        });
-
-        storeAndRestoreCarousel('overview-carousel');
-    }
-
     /**
      * Redraws the trend charts. Reads the last selected X-Axis type from the browser local storage and
      * redraws the trend charts.
      */
     function redrawTrendCharts () {
-        var isBuildOnXAxis = !(localStorage.getItem('#trendBuildAxis') === 'date');
+        const isBuildOnXAxis = !(localStorage.getItem('#trendBuildAxis') === 'date');
 
         /**
          * Creates a build trend chart that shows the number of issues for a couple of builds.
          * Requires that a DOM <div> element exists with the ID '#severities-trend-chart'.
          */
         view.getBuildTrend(isBuildOnXAxis, function (lineModel) {
-            $('#severities-trend-chart').renderTrendChart(lineModel.responseJSON, redrawTrendCharts);
+            renderZoomableTrendChart('severities-trend-chart', lineModel.responseJSON, redrawTrendCharts);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: ['renderZoomableTrendChart' is not defined. (no-undef)](https://app.codacy.com/app/uhafner/warnings-ng-plugin/pullRequest?prid=4437947)